CMakeLists.txt

```diff
@@ -1,70 +1,105 @@
+# Project settings
+#------------------------------------------------------------------------------#
 cmake_minimum_required(VERSION 3.9)
 
-##---------------------------------------------------------------------------##
-# Define your project name and set up major project options
-##---------------------------------------------------------------------------##
 project(Cabana LANGUAGES CXX)
-set(PROJECT_VERSION "0.3.0-dev")
+set(PROJECT_VERSION "1.0-dev")
 
+# If the user doesn't provide a build type default to release
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CXX_FLAGS)
   #release comes with -O3 by default
   set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." FORCE)
 endif(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CXX_FLAGS)
 
+# Use C++14
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF) #...without compiler extensions like gnu++11
 
+# Disable verbose makefiles
 option(CMAKE_VERBOSE_MAKEFILE "Generate verbose Makefiles" OFF)
+
+# use gnu standard install directories
 include(GNUInstallDirs)
 
+# point to cmake modules
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
+#------------------------------------------------------------------------------#
+# Dependencies
+#------------------------------------------------------------------------------#
 # find kokkos
 find_package(Kokkos 3.1 REQUIRED)
 
-# ensure that we can use lambdas
-if(Kokkos_ENABLE_CUDA)
-  kokkos_check(OPTIONS CUDA_LAMBDA)
-endif()
-
-option(Cabana_ENABLE_ARBORX "Build Cabana with ArborX support" OFF)
-if(Cabana_ENABLE_ARBORX)
-  find_package(ArborX REQUIRED)
-endif()
-
 # set supported kokkos devices
 set(CABANA_SUPPORTED_DEVICES SERIAL THREADS OPENMP CUDA HIP)
 
-# check user required device types
+# check user required kokkos device types
 foreach(_device ${CABANA_SUPPORTED_DEVICES})
   option(Cabana_REQUIRE_${_device} "Build Cabana with required Kokkos ${device} support" OFF)
   if( Cabana_REQUIRE_${_device} )
     kokkos_check( DEVICES ${_device} )
   endif()
 endforeach()
 
+# ensure that we can use lambdas
+if(Kokkos_ENABLE_CUDA)
+  kokkos_check(OPTIONS CUDA_LAMBDA)
+endif()
+
+# standard dependency macro
+macro(Cabana_add_dependency)
+  cmake_parse_arguments(CABANA_DEPENDENCY "" "PACKAGE" "" ${ARGN})
+  string(TOUPPER "${CABANA_DEPENDENCY_PACKAGE}" CABANA_DEPENDENCY_OPTION )
+  option(
+    Cabana_REQUIRE_${CABANA_DEPENDENCY_OPTION}
+    "Require Cabana to build with ${CABANA_DEPENDENCY_PACKAGE} support" OFF)
+  find_package( ${CABANA_DEPENDENCY_PACKAGE} QUIET )
+  if(Cabana_REQUIRE_${CABANA_DEPENDENCY_OPTION})
+    if( NOT ${CABANA_DEPENDENCY_OPTION}_FOUND )
+      message( FATAL_ERROR "${CABANA_DEPENDENCY_PACKAGE} required but not found" )
+    endif()
+  endif()
+  if ( ${CABANA_DEPENDENCY_OPTION}_FOUND )
+    message( STATUS "Found ${CABANA_DEPENDENCY_PACKAGE}" )
+    set(Cabana_HAVE_${CABANA_DEPENDENCY_OPTION} ON)
```

Comment:
where does that come from `<PackageName>_HAVE_<FOO>`?  I feel like I have seen it multiple places (like maybe in Trilinos) and every time I am thinking this is poor English.

Reply:
Yeah I agree the wording is weird - I have seen this somewhere in the past. I'm fine with changing it to `<PackageName>_ENABLE_<FOO>` or something similar. This variable is present so I can make a macro to use in header files in `#ifdef` statements.

Reply:
I have changed `HAVE` to `ENABLE`